api/v1beta1/azurecluster_default.go

```diff
@@ -232,17 +232,32 @@ func (c *AzureCluster) setAPIServerLBDefaults() {
 		if lb.Name == "" {
 			lb.Name = generateInternalLBName(c.ObjectMeta.Name)
 		}
-		if len(lb.FrontendIPs) == 0 {
-			lb.FrontendIPs = []FrontendIP{
-				{
-					Name: generateFrontendIPConfigName(lb.Name),
-					FrontendIPClass: FrontendIPClass{
-						PrivateIPAddress: DefaultInternalLBIPAddress,
-					},
-				},
-			}
+	}
+
+	// assign default private IP if not set
+	privateIPCount := 0
+	privateIPName := ""
+	for i := range lb.FrontendIPs {
+		if lb.FrontendIPs[i].FrontendIPClass.PrivateIPAddress != "" {
+			privateIPCount++
+			privateIPName = lb.FrontendIPs[i].Name
```

Comment:
Based on the default webhook, it looks like there should only be one privateIPName correct? In that case, maybe we can change `privateIPCount` to a boolean and break from this for loop if a private IP is found.

Reply:
added via 6b92b40cd654d61f71c3572a5f320c7ce8e8f314